### ISSUE TITLE
Important on Opacity

### ISSUE THREAD

**Issue by bradenhs** [typestyle/typestyle#133]
I'm just trying out typestyle on a new project and am loving it so far. I have hit one snag though. I'm trying to set the opacity as important like this:

```
style({
  opacity: '.5 !important'
})
```

But can't do it since typestyle expects `opacity` to be a number not a string. Is there another way to mark the opacity as important?

**Comment by basarat** [comment]
Options

* Use `opacity: '.5 !important' as any`
* OR (better) use csx.important: http://typestyle.io/#/utilities :rose:

**Comment by bradenhs** [comment]
Thanks! As far as I can tell `csx.important(0)` still returns a string though... so in either case I'll need `as any`. Is that correct?

**Comment by basarat** [comment]
> so in either case I'll need as any. Is that correct?

Yes. Sorry I didn't notice. Tagging in @notoriousb1t if you want to change it (don't have to. I leave the choice in your hands) :rose: https://github.com/typestyle/csx/blob/c0fdadd7f0dd0b3396d6bd49ecdcd04207fd35e0/src/strings.ts#L17

**Comment by notoriousb1t** [comment]
We could probably change important to lie about being the same type.  That would make more sense for important since it would preserve the intent.  Even though it is actually a string, for the purpose of CSS completion, it is a number.

As an aside, we need to update opacity to support global keywords:

/* Global values */
opacity: inherit;
opacity: initial;
opacity: unset;

**Comment by stefvw93** [comment]
Any updates on this?

**Comment by basarat** [comment]
@notoriousb1t How about `important<T>(x:T):T`?

# Current

```
export function important(val: number): string;
export function important<T extends string>(val: T): T;
export function important<T extends string>(val: number | T): string {
```
# New 

```
export function important<T>(val: T): T
```

**Comment by notoriousb1t** [comment]
@basarat  I think that is probably fine.  

It does mean important can wrap an {} and pretend it is still the same type, but I have a hard time believing someone would do that on accident.
